crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -2032,11 +2286,184 @@ mod test_shielded_wallet {
                 };
 
                 wallet.add_note(
-                    create_note(asset_data.clone(), rewardless, pa),
+                    create_note(asset_data, rewardless, pa),
                     vk,
                 );
-                let rewards_est = wallet.estimate_next_epoch_rewards(&context, &vk).await.expect("Test failed");
-                assert_eq!(rewards_est, 2 * reward_rate * i128::from(principal));
+                let balance = wallet
+                    .compute_shielded_balance(&vk)
+                    .await
+                    .unwrap()
+                    .unwrap_or_else(ValueSum::zero);
+                let rewards_est = wallet.estimate_next_epoch_rewards(&context, &balance)
+                    .await.expect("Test failed");
+                assert_eq!(
+                    rewards_est,
+                    DenominatedAmount::native(
+                        Amount::from_masp_denominated_i128(
+                            (1 + i128::from(shield_at_previous_epoch)) * reward_rate * i128::from(principal),
```

Comment:
I'm confused about the `1+` here.

Reply:
So the expected rewards are just `principal * reward_rate` but if we also shielded at the previous epoch then we need to multiply this value by 2, hence the first addition. You think it would be cleaner with two different assertions selected by an `if` on `shield_at_previous_epoch`?